lib/components/HiddenVisually/HiddenVisually.tsx

```diff
@@ -0,0 +1,29 @@
+import React, { useContext } from 'react';
+import { useStyles } from 'sku/react-treat';
+import { Box, BoxProps } from '../Box/Box';
+import TextContext from '../Text/TextContext';
+import HeadingContext from '../Heading/HeadingContext';
+import * as styleRefs from './HiddenVisually.treat';
+
+interface HiddenVisuallyProps {
+  children: BoxProps['children'];
+}
+
+export const HiddenVisually = ({ children }: HiddenVisuallyProps) => {
+  const styles = useStyles(styleRefs);
+  const inText = Boolean(useContext(TextContext));
+  const inHeading = Boolean(useContext(HeadingContext));
+
+  const component = inText || inHeading ? 'span' : 'div';
```

Comment:
Does this matter seeing as it's `position: absolute`?

Reply:
This is more about maintaining a semantic document rather than the visual layout.